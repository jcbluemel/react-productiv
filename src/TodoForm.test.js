import React, { Fragment } from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoForm from "./TodoForm";

const TEST_INTIAL_DATA = {
    title: "Test",
    description: "test",
    priority: 2
}

const mock = () => {};

it("renders without crashing", function () {
    render(<TodoForm initialFormData={TEST_INTIAL_DATA} handleSave={mock}/>);
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <TodoForm initialFormData={TEST_INTIAL_DATA} handleSave={mock} />);
    expect(asFragment()).toMatchSnapshot();
});

it("matches data-entered snapshot", function () {
    const { container } = render(
        <TodoForm handleSave={mock} />);

    const titleInput = container.querySelector("[name=title]");
    const descriptionInput = container.querySelector("[name=description]");
    const priorityInput = container.querySelector("[name=priority]");

    fireEvent.input(titleInput, { target: { value: "Test" } });
    fireEvent.input(descriptionInput, { target: { value: "test" } });
    fireEvent.input(priorityInput, { target: { value: 2 } });

    expect(container).toMatchSnapshot();
});

it("submitting form works", function () {
    const handleSaveMock = jest.fn();
    const { container, debug } = render(
        <TodoForm handleSave={handleSaveMock} />);

    const titleInput = container.querySelector("[name=title]");
    const descriptionInput = container.querySelector("[name=description]");
    const priorityInput = container.querySelector("[name=priority]");

    fireEvent.input(titleInput, { target: { value: "Test" } });
    fireEvent.input(descriptionInput, { target: { value: "test" } });
    fireEvent.input(priorityInput, { target: { value: 2 } });

    expect(handleSaveMock).toHaveBeenCalledTimes(0);
    fireEvent.click(container.querySelector(".NewTodoForm-addBtn"));
    expect(handleSaveMock).toHaveBeenCalledTimes(1);

    expect(container.querySelectorAll('input[value=""]')).toHaveLength(1);
    expect(container.querySelector('textarea')).toHaveTextContent('');
});

