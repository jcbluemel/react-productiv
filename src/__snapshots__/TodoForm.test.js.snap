// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches data-entered snapshot 1`] = `
<div>
  <form
    class="NewTodoForm"
  >
    <div
      class="mb-3"
    >
      <input
        aria-label="Title"
        class="form-control"
        id="newTodo-title"
        name="title"
        placeholder="Title"
        value="Test"
      />
    </div>
    <div
      class="mb-3"
    >
      <textarea
        aria-label="Description"
        class="form-control"
        id="newTodo-description"
        name="description"
        placeholder="Description"
      >
        test
      </textarea>
    </div>
    <div
      class="mb-3 d-flex justify-content-between"
    >
      <div
        class="w-75 d-flex justify-content-between"
      >
        <label
          class="d-inline-flex"
          for="newTodo-priority"
        >
          Priority:  
        </label>
        <select
          class="form-control form-control-sm d-inline-flex"
          id="newTodo-priority"
          name="priority"
        >
          <option
            value="1"
          >
            Ultra-Über
          </option>
          <option
            value="2"
          >
            Über
          </option>
          <option
            value="3"
          >
            Meh
          </option>
        </select>
      </div>
      <button
        class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
      >
        Gø!
      </button>
    </div>
  </form>
</div>
`;

exports[`matches data-entered snapshot 2`] = `
<div>
  <form
    class="NewTodoForm"
  >
    <div
      class="mb-3"
    >
      <input
        aria-label="Title"
        class="form-control"
        id="newTodo-title"
        name="title"
        placeholder="Title"
        value="Test"
      />
    </div>
    <div
      class="mb-3"
    >
      <textarea
        aria-label="Description"
        class="form-control"
        id="newTodo-description"
        name="description"
        placeholder="Description"
      >
        test
      </textarea>
    </div>
    <div
      class="mb-3 d-flex justify-content-between"
    >
      <div
        class="w-75 d-flex justify-content-between"
      >
        <label
          class="d-inline-flex"
          for="newTodo-priority"
        >
          Priority:  
        </label>
        <select
          class="form-control form-control-sm d-inline-flex"
          id="newTodo-priority"
          name="priority"
        >
          <option
            value="1"
          >
            Ultra-Über
          </option>
          <option
            value="2"
          >
            Über
          </option>
          <option
            value="3"
          >
            Meh
          </option>
        </select>
      </div>
      <button
        class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
      >
        Gø!
      </button>
    </div>
  </form>
</div>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <div
      class="mb-3"
    >
      <input
        aria-label="Title"
        class="form-control"
        id="newTodo-title"
        name="title"
        placeholder="Title"
        value="Test"
      />
    </div>
    <div
      class="mb-3"
    >
      <textarea
        aria-label="Description"
        class="form-control"
        id="newTodo-description"
        name="description"
        placeholder="Description"
      >
        test
      </textarea>
    </div>
    <div
      class="mb-3 d-flex justify-content-between"
    >
      <div
        class="w-75 d-flex justify-content-between"
      >
        <label
          class="d-inline-flex"
          for="newTodo-priority"
        >
          Priority:  
        </label>
        <select
          class="form-control form-control-sm d-inline-flex"
          id="newTodo-priority"
          name="priority"
        >
          <option
            value="1"
          >
            Ultra-Über
          </option>
          <option
            value="2"
          >
            Über
          </option>
          <option
            value="3"
          >
            Meh
          </option>
        </select>
      </div>
      <button
        class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
      >
        Gø!
      </button>
    </div>
  </form>
</DocumentFragment>
`;
